### PR TITLE
[BUGFIX] GLUE-764 Wrong uuid "cart" behavior when adding an item to cart

### PR DESCRIPTION
**Pull request by AsonUnique**
- Developer(s): @AsonUnique

- Ticket: https://spryker.atlassian.net/browse/GLUE-764

- Academy PR: -

#### Please confirm

- [x] No new OS components - or they have been approved by legal department

#### Documentation

- [x] Functional documentation provided or in progress?
- [x] Integration guide for projects provided or not needed?
- [x] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   UuidBehavior          | patch                 |                       |

#### Release Notes

Fixed generated UUID duplication bug.

-----------------------------------------

#### Module UuidBehavior

_**Patch:** Backwards-compatible bug fix_

Def of done (by responsible developer):
- [x] All changes are backward-compatible. Outdated code is marked as deprecated.
- [x] The change is isolated and not mixed with any cleanups or other changes.
- [x] New and changed facade methods covered by functional tests. / Has nothing to test.
- [x] New and changed complex business logic is covered by unit or integration tests. / Has nothing to test.

##### Change log

Bugfixes

Fixed generated UUID duplication bug.

### CODE REVIEW

File: src/Spryker/Zed/UuidBehavior/Persistence/Propel/Behavior/UuidBehavior.php [hunk 1]
```diff
@@ -23,6 +23,11 @@ class UuidBehavior extends Behavior
     protected const ERROR_INVALID_KEY_COLUMNS_FORMAT = 'Invalid data passed to %s as "key_columns" parameter';
     protected const ERROR_COLUMN_NOT_FOUND = 'Column %s that is specified for generating UUID is not exist.';
 
+    /**
```
Comment (dereuromark) [inline src/Spryker/Zed/UuidBehavior/Persistence/Propel/Behavior/UuidBehavior.php]: {inheritDoc} would be nice here

**Review by ghost — APPROVED** [review]

**Review by limeeugenia — APPROVED** [review]

**Review by valerio8787 — APPROVED** [review]
@AsonUnique Please check my changes
Prepare correct PR template
Prepare bc-check branch

**Review by tamasnyulas — APPROVED** [review]

File: src/Spryker/Zed/UuidBehavior/Persistence/Propel/Behavior/templates/objectUpdateUuidAfterInsert.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
+/**
+ * @param ConnectionInterface $con
+ *
+ * @return void
+ */
+protected function updateUuidAfterInsert(ConnectionInterface $con = null)
+{
+    if (empty($this->getUuid())) {
```
Comment (dereuromark) [inline src/Spryker/Zed/UuidBehavior/Persistence/Propel/Behavior/templates/objectUpdateUuidAfterInsert.php]: Please do not use empty with methods. Use just

    if (!$this->getUuid()) 

etc

File: src/Spryker/Zed/UuidBehavior/Persistence/Propel/Behavior/templates/objectUpdateUuidBeforeUpdate.php [hunk 3]
```diff
@@ -0,0 +1,10 @@
+
+/**
+ * @return void
+ */
+protected function updateUuidBeforeUpdate()
+{
+    if (empty($this->getUuid())) {
```
Comment (dereuromark) [inline src/Spryker/Zed/UuidBehavior/Persistence/Propel/Behavior/templates/objectUpdateUuidBeforeUpdate.php]: Please do not use empty with methods. Use just

    if (!$this->getUuid()) 

etc

**Review by dereuromark — APPROVED** [review]

File: src/Spryker/Zed/UuidBehavior/Persistence/Propel/Behavior/Exception/MissingAttributeException.php [hunk 4]
```diff
@@ -1,13 +0,0 @@
...
-/**
- * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
- * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
- */
-
-namespace Spryker\Zed\UuidBehavior\Persistence\Propel\Behavior\Exception;
-
-use Propel\Runtime\Exception\PropelException;
-
-class MissingAttributeException extends PropelException
```
Comment (dereuromark) [inline src/Spryker/Zed/UuidBehavior/Persistence/Propel/Behavior/Exception/MissingAttributeException.php:11]: Please note that removing a class is usually NOT BC.
Better to deprecate when you release a patch.

**Pull request merged**